ITR/interfaces.py

```diff
@@ -165,30 +207,77 @@ def __modify_schema__(cls, field_schema):
         ),
     )
 
+
+class EI_Quantity(BaseModel):
+
+    dims_list: List[str]
+
+    @validator('dims_list')
+    def units_must_be_registered(cls, v):
+        for d in v:
+            try:
+                registry.get_dimensionality(d)
+            except KeyError:
+                raise ValueError(f"{d} is not a valid dimensionality in pint!")
+        return v
+
+    class Config:
+        validate_assignment = True
+        schema_extra = schema_extra
+        json_encoders = {
+            Quantity: str,
+        }
+
+def ei_quantity(dims_list: List[str]) -> type:
+    """A method for making a pydantic compliant Pint Emissions Intensity (EI) quantity."""
+
+    try:
+        for dimensionality in dims_list:
+            registry.get_dimensionality(dimensionality)
+    except KeyError:
+        raise ValueError(f"{dimensionality} is not a valid dimensionality in pint!")
+
+    @classmethod
+    def __get_validators__(cls):
+        yield cls.validate
+
+    @classmethod
+    def validate(cls, value):
+        if isinstance(value, dict):
+            quantity = Q_(value['units'])
+        else:
+            quantity = Q_(value)
+        for dimensionality in EI_Quantity(dims_list=dims_list).dims_list:
+            if quantity.check(dimensionality):
+                return quantity
+        raise DimensionalityError(value.units, f"in [{EI_Quantity(dims_list).dims_list}]")
+
+    @classmethod
+    def __modify_schema__(cls, field_schema):
+        field_schema.update(
+            {"$ref": "#/definitions/EI_Quantity"}
+        )
+    
+    return type(
+        "EI_Quantity",
+        (EI_Quantity,),
+        dict(
+            __get_validators__=__get_validators__,
+            __modify_schema__=__modify_schema__,
+            dims_list=f"List[str] = {dims_list}",
+            validate=validate,
+        ),
+    )
+
 # FIXME: delete this when we are done converting to new Pint
 class ProductionMetric(BaseModel):
     units: str
     @validator('units')
     def unit_must_be_production(cls, v):
         qty = ureg(v)
-        if qty.is_compatible_with("Wh"):
-            return v
-        if qty.is_compatible_with("Fe_ton"):
-            return v
-        if qty.is_compatible_with("passenger km"):
-            return v
-        if qty.is_compatible_with("tkm"):
-            return v
-        if qty.is_compatible_with("boe"):
-            return v
-        if qty.is_compatible_with("t Aluminum"):
-            return v
-        if qty.is_compatible_with("t Cement"):
-            return v
-        if qty.is_compatible_with("USD"):
-            return v
-        if qty.is_compatible_with("m**2"):
-            return v
+        for pu in _production_units:
```

Comment:
where do we define `_production_units`?

Reply:
Line 36 in the current GitHub branch (template-v2) of the MichaelTiemannOSC/ITR repository.

Reply:
all right, then, the question is the same:
`interfaces.py` is a collection of classes.
Shouldn't we avoid methods w/o classes in such situations?
why can't it be a property of some appropriate class within `interfaces.py`?

Reply:
`intefaces.py` module contains classes.
But additionally, it has functions, which live outside of classes.
Should we turn them into class methods?